acme/acme/client.py

```diff
@@ -66,15 +66,13 @@ def __init__(self, directory, key, alg=jose.RS256, verify_ssl=True,
     @classmethod
     def _regr_from_response(cls, response, uri=None, new_authzr_uri=None,
                             terms_of_service=None):
-        terms_of_service = (
-            response.links['terms-of-service']['url']
-            if 'terms-of-service' in response.links else terms_of_service)
+        if 'terms-of-service' in response.links:
+            terms_of_service = response.links['terms-of-service']['url']
+        if 'next' in response.links:
+            new_authzr_uri = response.links['next']['url']
 
         if new_authzr_uri is None:
-            try:
-                new_authzr_uri = response.links['next']['url']
-            except KeyError:
-                raise errors.ClientError('"next" link missing')
+            raise errors.ClientError(response, 'missing "new_authrz_uri"')
```

Comment:
What is the issue with the response argument?  I figured having a bit more context in the exception is helpful (e.g., to have the response headers).  That is why I added the explicit field in 2085.

I am happy with changing the message, but from my point of view an API user is actually more familiar with new_authrz_uri (as it is the field name in the object) than with the next/url notation (as the API hides most of the http call details).  So just trying to figure out the underlying principe for messaging for the future.